### ISSUE TITLE
implement `weights` argument for `slab()`

### ISSUE THREAD

**Issue by brownag** [ncss-tech/aqp#229]
A planned feature for `slab()` is to (re-)support site-level weights. See #3

When specifying some dummy weights the following error message is issued
``` r
library(aqp)
#> This is aqp 1.30
#> 
#> Attaching package: 'aqp'
#> The following object is masked from 'package:stats':
#> 
#>     filter
data(sp1)
depths(sp1) <- id ~ top + bottom
slab(sp1, fm = ~ prop, weights = 1:length(sp1))
#> Error: weighted aggregation is not yet supported
```

Example use case: 
 - component percent weighted averages from an SPC derived from one or more SSURGO mapunits worth of horizon data (containing multiple soil components). 
 
 See https://github.com/ncss-tech/soilDB/pull/194 for some sample DMU record IDs for weighted average aggregation and validation against Web Soil Survey aggregation

**Comment by dylanbeaudette** [comment]
Thanks, this is described in #3.

**Comment by brownag** [comment]
Note to consider option of supplying weights as a character vector length 1 column name, or as a vector equal in length to number of profiles.
In principle I think it would be possible to support a matrix of weights, and/or horizon level weights.

**Comment by dylanbeaudette** [comment]
> Note to consider option of supplying weights as a character vector length 1 column name (implemented in #258), or as a vector equal in length to number of profiles. In principle I think it would be possible to support a matrix of weights, and/or horizon level weights.

Yes: the original implementation (2008?) allowed for site-level or horizon-level weights. It wasn't entirely clear to me at the time how to implement horizon-level weighting into things like weighted SD over intervals > 1cm thickness.

I'd prefer that the updates as part of this issue and related PR implement site-level weighted first, via named column from site. Once the details of horizon-level weighting are determined, then I'd prefer to use a similar approach via named column from horizons.

**Comment by brownag** [comment]
> I'd prefer that the updates as part of this issue and related PR implement site-level weighted first, via named column from site. 

This is done
